src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -255,4 +269,99 @@ private static Stream<Arguments> fuzzyLowCount() {
         arguments(20, 20),
         arguments(21, 21));
   }
+
+  private <T> void mockRunQueryForHierarchy(
+      CloudPlatform cloudPlatform, List<T> results, Dataset dataset) {
+    CloudPlatformWrapper.of(cloudPlatform)
+        .choose(
+            () ->
+                when(bigQueryDatasetPdao.runQuery(
+                        any(),
+                        any(),
+                        argThat(
+                            (ArgumentMatcher<BigQueryDatasetPdao.Converter<T>>) Objects::nonNull)))
+                    .thenReturn(results),
+            () ->
+                when(azureSynapsePdao.runQuery(
+                        any(),
+                        argThat((ArgumentMatcher<AzureSynapsePdao.Converter<T>>) Objects::nonNull)))
+                    .thenReturn(results));
+  }
+
+  private static void assertParentQueryResult(SnapshotBuilderService.ParentQueryResult result) {
+    assertThat(result.parentId(), is(1));
+    assertThat(result.childId(), is(2));
+    assertThat(result.childName(), is("name"));
+  }
+
+  @Test
+  void testParentQueryResult() throws Exception {
+    var resultSet = mock(ResultSet.class);
+    when(resultSet.getInt(HierarchyQueryBuilder.PARENT_ID)).thenReturn(1);
+    when(resultSet.getInt(HierarchyQueryBuilder.CONCEPT_ID)).thenReturn(2);
+    when(resultSet.getString(HierarchyQueryBuilder.CONCEPT_NAME)).thenReturn("name");
+    assertParentQueryResult(new SnapshotBuilderService.ParentQueryResult(resultSet));
+
+    var fieldValueList =
+        FieldValueList.of(
+            List.of(
+                FieldValue.of(FieldValue.Attribute.PRIMITIVE, "1"),
+                FieldValue.of(FieldValue.Attribute.PRIMITIVE, "2"),
+                FieldValue.of(FieldValue.Attribute.PRIMITIVE, "name")),
+            Field.of(HierarchyQueryBuilder.PARENT_ID, StandardSQLTypeName.NUMERIC),
+            Field.of(HierarchyQueryBuilder.CONCEPT_ID, StandardSQLTypeName.NUMERIC),
+            Field.of(HierarchyQueryBuilder.CONCEPT_NAME, StandardSQLTypeName.STRING));
+
+    assertParentQueryResult(new SnapshotBuilderService.ParentQueryResult(fieldValueList));
+  }
+
+  static SnapshotBuilderConcept concept(String name, int id) {
```

Comment:
Nit: Would you move this up so we have clumped the static methods and the tests in groups?

Reply:
My usual convention is to put methods like this above the first usage in the class, unless there's another ordering going on. That way it's close to where it's used and before it, in case a reader is reading the code from top to bottom.

Are you suggesting it should go after `assertParentQueryResult()` instead?

Reply:
I could move `mockRunQueryForHierarchy()` down here, that would be a little better.

Reply:
I'm fine with that as the approach.